src/core/dbus-service.c

```diff
@@ -55,7 +55,7 @@ static int property_get_exit_status_set(
                 return r;
 
         SET_FOREACH(id, status_set->status, i) {
-                int val = PTR_TO_INT(id);
+                int32_t val = PTR_TO_INT32(id);
```

Comment:
hmm, the `status_set->status` `Set` object actually *does* contain `int` values, hence this should be:

```c
int32_t val = PTR_TO_INT(id);
``` 

i think, as we first need to extract and convert the int value from the Set, and then convert it here to in32_t so that we can use it for dbus marshalling.

Reply:
Indeed. I was so foolish...